attachments/models.py

```diff
@@ -232,8 +235,15 @@ def is_valid(self):
     @property
     def upload_forms(self):
         from .forms import PropertyForm
+        invalid_uploads = []
         for upload in self.uploads.all():
-            yield upload, self._forms.get(upload, PropertyForm(instance=upload))
+            try:
+                yield None, upload, self._forms.get(upload, PropertyForm(instance=upload))
+            except ValidationError as ex:
+                invalid_uploads.append(upload.file_name)
+                yield ex.message, None, None
+        self.uploads.filter(file_name__in=invalid_uploads).delete()
```

Comment:
Why not keep a list of the actual `Upload` objects and delete them, instead of doing a `file_name__in` which is slower (and could potentially delete more than you intend, especially if 2 files of different types were uploaded with the same name, and one was flagged and the other wasn't).